content/400-reference/200-api-reference/050-prisma-client-reference.mdx

```diff
@@ -0,0 +1,3253 @@
+
+<details><summary>Overview: Race conditions</summary>
+
+A race conditions occurs when two or more operations must be done in sequence in order to complete a task. In the following example, two clients try to increase the same field (`postCount`) by one:
+
+| Client   | Operation           | Value  |
+| :------- | :------------------ | :----- |
+| Client 1 | **Get** field value | `21`   |
+| Client 2 | **Get** field value | `21`   |
+| Client 2 | **Set** field value | `22`   |
+| Client 1 | **Set** field value | `22` ✘ |
+
+The value _should_ be `23`, but the two clients did not read and write to the `postCount` field in sequence. Atomic operations on update combine read and write into a single operation, which prevents a race condition:
+
+| Client   | Operation                   | Value              |
+| :------- | :-------------------------- | :----------------- |
+| Client 1 | **Get and set** field value | `21` &rarr; `22`   |
+| Client 2 | **Get and set** field value | `22` &rarr; `23` ✔ |
+
+</details>
+
+### Operators
+
+| Option      | Description                                                   |
+| :---------- | :------------------------------------------------------------ |
+| `increment` | Adds `n` to the current value.                                |
+| `decrement` | Subtacts `n` from the current value.                          |
+| `multiply`  | Multiplies the current value by `n`.                          |
+| `divide`    | Divides the current value by `n`.                             |
+| `set`       | Sets the current field value. Identical to `{ myField : n }`. |
+
+### Reference
+
+```ts file=index.d.ts
+export type IntFieldUpdateOperationsInput = {
+  set?: number
+  increment?: number
+  decrement?: number
+  multiply?: number
+  divide?: number
+}
+```
+
+### Remarks
+
+- You can only perform **one** atomic update per field, per query.
+
+### Examples
+
+#### Increment all `view` and `likes` fields of all `Post` records by `1`
+
+```ts
+const updatePosts = await prisma.post.updateMany({
+  data: {
+    views: {
+      increment: 1,
+    },
+    likes: {
+      increment: 1,
+    },
+  },
+})
+```
+
+#### Set all `count` fields of all `Post` records to `0`
+
+```ts
+const updatePosts = await prisma.post.updateMany({
+  data: {
+    views: {
+      set: 0,
+    },
+  },
+})
+```
+
+Can also be written as:
+
+```ts
+const updatePosts = await prisma.post.updateMany({
+  data: {
+    views: 0,
+  },
+})
+```
+
+## Client methods
+
+Client-level methods are prefixed by `$`.
+
+<!-- prettier-ignore -->
+### <inlinecode>$disconnect()</inlinecode>
+
+The `$disconnect()` method closes the database connections that were established when `$connect` was called and stops the process that was running Prisma's query engine. See [Connection management for an overview of `$connect()` and `$disconnect()`](../../..//concepts/components/prisma-client/connection-management).
```

Comment:
IMO closing the link a bit earlier makes this look cleaner/less noisy

```suggestion
The `$disconnect()` method closes the database connections that were established when `$connect` was called and stops the process that was running Prisma's query engine. See [Connection management](../../..//concepts/components/prisma-client/connection-management) for an overview of `$connect()` and `$disconnect()`.
```

![image](https://user-images.githubusercontent.com/4058327/105839181-64dde780-5fd1-11eb-83d1-cc5cb25f2127.png)